### PR TITLE
Support Custom HTML in Workflow Notes

### PR DESCRIPTION
**Pull request by andreas-gruenwald**
This PR allows the configuration of renderers to dynamically output HTML markup in Workflow notes. 

This is useful to show summaries and alike, before the user is actually going to commit the workflow change.

Example: 
image

Notes can be placed on three different positions (top, center, bottom) for both transitions, and global actions.
A custom renderer service and a default position can be configured in the workflow configuration yml.

### CODE REVIEW

File: lib/Workflow/Notes/NotesAwareTrait.php [hunk 1]
```diff
@@ -59,4 +62,22 @@ public function getNotesAdditionalFields(): array
     {
         return $this->options['notes']['additionalFields'] ?? [];
     }
+
+    /**
+     * Inject service via compiler pass.
+     * @param $customHtmlService
+     */
+    public function setCustomHtmlService($customHtmlService) {
+        if ($customHtmlService instanceof AbstractCustomHtmlService) {
```
Comment (dpfaffenbauer) [inline lib/Workflow/Notes/NotesAwareTrait.php:71]: instanceof CustomHtmlServiceInterface

Comment (dpfaffenbauer) [reply lib/Workflow/Notes/NotesAwareTrait.php:71]: sorry, no, actually that is fine.... But the AbstractCustomHtmlService should be a interface with it's extra functions, right?

Comment (andreas-gruenwald) [reply lib/Workflow/Notes/NotesAwareTrait.php:71]: You mean an additional interface for ``getTransitionName()``, ``getActionName()``, ``getPosition()`` and so forth?
Don't know if this is necessary, as those fields are intended to be mainly used within the class.

Comment (dpfaffenbauer) [reply lib/Workflow/Notes/NotesAwareTrait.php:71]: not sure honestly... I always use separate interfaces, also for this use-cases. What if a custom Symfony (not Pimcore) Workflow tries to do this?

File: doc/Development_Documentation/07_Workflow_Management/01_Configuration_Details/README.md [hunk 2]
```diff
@@ -222,7 +222,16 @@ pimcore:
 
                                     # Will be passed to the underlying Pimcore data object field type. Can be used to configure the options of a select box for example.
                                     fieldTypeSettings:    []
-
+                                    
+                            # Configure the output of custom HTML for a transition.
+                            customHtml:
+                            
+                                # Define a custom service for rendering custom HTML within the note modal.
+                                service: 'AppBundle\Service\Workflow\CustomHtmlService'
```
Comment (dvesh3) [inline doc/Development_Documentation/07_Workflow_Management/01_Configuration_Details/README.md:230]:
```suggestion
                                service: 'App\Service\Workflow\CustomHtmlService'
```
please see https://github.com/pimcore/demo/commit/beb4a68c76351f17a6f048b30f71899da66be0a3#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R24

File: doc/Development_Documentation/07_Workflow_Management/01_Configuration_Details/README.md [hunk 3]
```diff
@@ -298,4 +307,13 @@ pimcore:
                                 required:             false
                                 setterFn:             ~
                                 fieldTypeSettings:    []
+                                
+                        # Configure the output of custom HTML for a transition.
+                        customHtml:
+                        
+                            # Define a custom service for rendering custom HTML within the note modal.
+                            service: 'AppBundle\Service\Workflow\CustomHtmlService'
```
Comment (dvesh3) [inline doc/Development_Documentation/07_Workflow_Management/01_Configuration_Details/README.md:315]:
```suggestion
                            service: 'App\Service\Workflow\CustomHtmlService'
```

**Comment by dvesh3** [comment]
@chilladelia thanks a lot!

**Comment by andreas-gruenwald** [comment]
Thanks for merging!

**Pull request merged**